tests/bluetooth/tester/src/btp_tmap.c

```diff
@@ -0,0 +1,141 @@
+/* btp_tmap.c - Bluetooth TMAP Tester */
+
+/*
+ * Copyright (c) 2023 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/bap.h>
+#include <zephyr/bluetooth/audio/tmap.h>
+#include <zephyr/bluetooth/audio/pacs.h>
+#include <zephyr/bluetooth/services/ias.h>
+
+#include "btp/btp.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(bttester_tmap, CONFIG_BTTESTER_LOG_LEVEL);
+
+#define TMAP_CG_SUPPORTED                                                                          \
+	(IS_ENABLED(CONFIG_BT_CAP_INITIATOR) && IS_ENABLED(CONFIG_BT_BAP_UNICAST_CLIENT) &&        \
+	 IS_ENABLED(CONFIG_BT_TBS) && IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR))
+
+#define TMAP_CT_SUPPORTED                                                                          \
+	(IS_ENABLED(CONFIG_BT_CAP_ACCEPTOR) && IS_ENABLED(CONFIG_BT_BAP_UNICAST_SERVER) &&         \
+	 IS_ENABLED(CONFIG_BT_TBS_CLIENT) &&                                                       \
+	 (IS_ENABLED(CONFIG_BT_ASCS_ASE_SNK) &&                                                    \
+	  IS_ENABLED(CONFIG_BT_VCP_VOL_REND) == IS_ENABLED(CONFIG_BT_ASCS_ASE_SNK)))
+
+#define TMAP_UMS_SUPPORTED                                                                         \
+	(IS_ENABLED(CONFIG_BT_CAP_INITIATOR) &&                                                    \
+	 IS_ENABLED(CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK) && IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR) && \
+	 IS_ENABLED(CONFIG_BT_MCS))
+
+#define TMAP_UMR_SUPPORTED                                                                         \
+	(IS_ENABLED(CONFIG_BT_CAP_ACCEPTOR) && IS_ENABLED(CONFIG_BT_ASCS_ASE_SNK) &&               \
+	 IS_ENABLED(CONFIG_BT_VCP_VOL_REND))
+
+#define TMAP_BMS_SUPPORTED                                                                         \
+	(IS_ENABLED(CONFIG_BT_CAP_INITIATOR) && IS_ENABLED(CONFIG_BT_BAP_BROADCAST_SOURCE))
+
+#define TMAP_BMR_SUPPORTED                                                                         \
+	(IS_ENABLED(CONFIG_BT_CAP_ACCEPTOR) && IS_ENABLED(CONFIG_BT_BAP_BROADCAST_SINK))
```

Comment:
This is a duplicate from the shell, right?

I think it makes sense to move this to `tmap.h` so that they are generally available (but can be done in a separate PR)

Reply:
You're right. I create new PR 
https://github.com/zephyrproject-rtos/zephyr/pull/68742